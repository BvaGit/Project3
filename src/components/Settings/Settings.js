import React from 'react';
import { useState, useEffect } from 'react';

import { DEFAULT_LOCALE } from '../../constants/locale';
import { DEFAULT_THEME, THEME } from '../../constants/theme';
import CustomModal from '../CustomModal';
import ModalSettings from '../ModalSettings';

import './settings.scss'

function Settings(props) {
    const [isOpen, setIsOpen] = useState(false);

    const handleClick = () => {
        setIsOpen(prev => !prev);
    };

    const handleTheme = (selectedTheme) => {
        if (selectedTheme === THEME.LIGHT) {
            document.getElementById('root').classList.remove('theme-dark');
            document.getElementById('portal').classList.remove('theme-dark');
        } else {
            document.getElementById('root').classList.add('theme-dark');
            document.getElementById('portal').classList.add('theme-dark');
        }
    };

    useEffect(() => {
        const { account, handleLanguage, changeAccountField } = props;

        if (!account.locale) {
            changeAccountField({ locale: DEFAULT_LOCALE });
        }
        if (!account.theme) {
            changeAccountField({ theme: DEFAULT_THEME });
        }
        handleLanguage(account.locale ? account.locale : DEFAULT_LOCALE);
        handleTheme(account.theme ? account.theme : DEFAULT_THEME);
    }, [props.account]);
    
    return(
        <div>
            <img onClick={handleClick} className="rooms-header__settings" src="../../../public/assets/images/seting_2.png"/>
            {
                isOpen && 
                <CustomModal 
                    header="Settings" 
                    handlecloseModal={handleClick}
                    content={
                        <ModalSettings 
                            closeModal={handleClick} 
                            handleLanguage={props.handleLanguage}
                            handleTheme={handleTheme}
                            languageValue={props.languageValue}
                            changeAccountField={props.changeAccountField}
                            updateAccount={props.updateAccount}
                            account={props.account}
                        />
                    }    
                />
            }
        </div>
    )
}

export default Settings
