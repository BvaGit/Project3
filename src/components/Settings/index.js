import { connect } from 'react-redux';

import Settings from "./Settings";
import { handleLanguage } from '../../store/localization/actions';
import { changeMyAccountField } from '../../store/user/actions';
import { updateMyAccount } from '../../store/user/asyncActions';
import { getUserMyAccount } from '../../store/user/selectors';

const mapDispatchToProps = (dispatch) => ({
    handleLanguage: (payload) => dispatch(handleLanguage(payload)),
    updateAccount: () => dispatch(updateMyAccount()),
    changeAccountField: (data) => dispatch(changeMyAccountField(data)),
})

const mapStateToProps = (state) => ({
    account: getUserMyAccount(state)
});

export default connect(mapStateToProps, mapDispatchToProps)(Settings);
