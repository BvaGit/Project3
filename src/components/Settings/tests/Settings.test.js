import React from 'react';
import { shallow, mount } from 'enzyme';
import Settings from '../Settings';


describe('Settings', () => {
    it("Should match snapshot", () => {
        const component = shallow(<Settings />)
        expect(component.html()).toMatchSnapshot()
    })
})