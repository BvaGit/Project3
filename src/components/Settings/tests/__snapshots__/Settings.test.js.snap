// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Should match snapshot 1`] = `"<div><img class=\\"rooms-header__settings\\" src=\\"../../../public/assets/images/seting_2.png\\"/></div>"`;
